### PR TITLE
Backend: remove `enterprise/internal`

### PR DESCRIPTION
**Pull request by camdencheek**
## Test plan

It builds. Mostly a bunch of file shuffling.

### CODE REVIEW

File: cmd/frontend/auth/non_public_test.go [hunk 1]
```diff
@@ -14,13 +14,12 @@ import (
 	"github.com/sourcegraph/sourcegraph/internal/actor"
 	"github.com/sourcegraph/sourcegraph/internal/conf"
 	"github.com/sourcegraph/sourcegraph/internal/database"
-	"github.com/sourcegraph/sourcegraph/internal/search/job/jobutil"
 	"github.com/sourcegraph/sourcegraph/lib/pointers"
 	"github.com/sourcegraph/sourcegraph/schema"
 )
 
 func TestAllowAnonymousRequest(t *testing.T) {
-	ui.InitRouter(database.NewMockDB(), jobutil.NewUnimplementedEnterpriseJobs())
```
Comment (camdencheek) [inline cmd/frontend/auth/non_public_test.go:23]: @eseliger sorry to delete all your hard work here 😄

Comment (eseliger) [reply cmd/frontend/auth/non_public_test.go:23]: I'm more than happy to see this go, I didn't want to build it in the first place 😬

File: migrations/frontend/squashed.sql [hunk 2]
```diff
@@ -2760,7 +2760,7 @@ CREATE TABLE insights_query_runner_jobs (
     trace_id text
 );
 
-COMMENT ON TABLE insights_query_runner_jobs IS 'See [enterprise/internal/insights/background/queryrunner/worker.go:Job](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:enterprise/internal/insights/background/queryrunner/worker.go+type+Job&patternType=literal)';
+COMMENT ON TABLE insights_query_runner_jobs IS 'See [internal/insights/background/queryrunner/worker.go:Job](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:internal/insights/background/queryrunner/worker.go+type+Job&patternType=literal)';
```
Comment (eseliger) [inline migrations/frontend/squashed.sql:2763]: will this cause drift?

**Review by eseliger — APPROVED** [review]
Looks solid to me, I briefly looked at all the files, but trust you that the 634 file diff will be good :D Happy to see the search enterprise thing go away!

File: migrations/frontend/squashed.sql (hunk 2 above)
Comment (camdencheek) [reply migrations/frontend/squashed.sql:2763]: Sniped by automerge. I didn't even think about a comment causing drift...I'm not sure.

Comment (eseliger) [reply migrations/frontend/squashed.sql:2763]: asked dr database, this shouldn't be a problem if CI didn't complain :)

Comment (camdencheek) [reply migrations/frontend/squashed.sql:2763]: Cool. I also ran `sg migration drift` on main to see if it complained and got nothing

Comment (camdencheek) [reply migrations/frontend/squashed.sql:2763]: Thanks for checking!

**Pull request merged**